File: spring-cloud-integrated-demo/user-service/src/main/java/com/github/jccode/springcloud/integrateddemo/user/service/UserService.java
package com.github.jccode.springcloud.integrateddemo.user.service;

import com.github.jccode.springbootsample.core.repo.CrudMapper;
import com.github.jccode.springbootsample.core.service.CrudServiceImpl;
import com.github.jccode.springcloud.integrateddemo.user.model.User;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class UserService extends CrudServiceImpl<User> {

    @Autowired
    public UserService(CrudMapper<User> mapper) {
        super(mapper);
    }

}

File: spring-boot-rest-demo/src/main/scala/com/github/jccode/springbootrestdemo2/service/UserService.scala
package com.github.jccode.springbootrestdemo2.service

import com.github.jccode.springbootrestdemo2.model.User
import org.springframework.stereotype.Service

@Service
class UserService {

  val userMap = Map(
    "hello" -> User(1, "hello", "world", 10),
    "tom" -> User(2, "tom", "cat", 10),
  )

  def find(name: String): User = userMap(name)
}
